src/ja/user/message/add.py

```diff
@@ -43,6 +44,21 @@ def execute(self, database: ServerDatabase) -> Response:
         if db_job_id is not None:
             return Response(result_string="Job with id %s already exists" % job.uid,
                             is_success=False)
+
+        if job.scheduling_constraints.special_resources is not None:
```

Comment:
When is this None? In tests?

Reply:
It seems I misremembered whether special_resources is allowed to be None.
I re-ran unittests without this line and everything was working as intended.